compiler/pathutils.nim

```diff
@@ -22,25 +28,39 @@ type
 proc isEmpty*(x: AnyPath): bool {.inline.} = x.string.len == 0
 
 proc copyFile*(source, dest: AbsoluteFile) =
-  os.copyFile(source.string, dest.string)
+  when defined(nodejs):
+    fs.createReadStream(source.cstring).pipe(fs.createWriteStream(dest.cstring));
+  else:
+    os.copyFile(source.string, dest.string)
```

Comment:
but this is the wrong place to do this; instead make `os.copyFile` with with `nodejs`
likewise with other procs.

That way is much more useful as it'll make clients of os work with nodejs